yarn-project/simulator/src/public/transitional_adaptors.ts

```diff
@@ -91,145 +70,23 @@ export function createPublicExecutionContext(avmContext: AvmContext, calldata: F
-
-  const returnValues = result.output;
-
-  // TODO: Support nested executions.
-  const nestedExecutions: PublicExecutionResult[] = [];
-  // TODO keep track of side effect counters
-  const startSideEffectCounter = Fr.ZERO;
-  const endSideEffectCounter = Fr.ZERO;
-
-  return {
-    execution,
-    nullifierReadRequests,
-    nullifierNonExistentReadRequests,
-    newNoteHashes,
-    newL2ToL1Messages,
-    startSideEffectCounter,
-    endSideEffectCounter,
-    newNullifiers,
-    contractStorageReads,
-    contractStorageUpdateRequests,
-    returnValues,
-    nestedExecutions,
-    unencryptedLogs,
-    reverted: result.reverted,
-    revertReason: result.revertReason ? createSimulationError(result.revertReason) : undefined,
-    gasLeft: endMachineState.gasLeft,
-  };
 }
 
 export function convertPublicExecutionResult(res: PublicExecutionResult): AvmContractCallResults {
   return new AvmContractCallResults(res.reverted, res.returnValues, res.revertReason);
 }
 
 export function updateAvmContextFromPublicExecutionResult(ctx: AvmContext, result: PublicExecutionResult): void {
-  // We have to push these manually and not use the trace* functions
-  // so that we respect the side effect counters.
-  for (const readRequest of result.contractStorageReads) {
-    ctx.persistableState.trace.publicStorageReads.push({
-      storageAddress: ctx.environment.storageAddress,
-      exists: true, // FIXME
-      slot: readRequest.storageSlot,
-      value: readRequest.currentValue,
-      counter: new Fr(readRequest.sideEffectCounter ?? Fr.ZERO),
-    });
-  }
-
   for (const updateRequest of result.contractStorageUpdateRequests) {
-    ctx.persistableState.trace.publicStorageWrites.push({
-      storageAddress: ctx.environment.storageAddress,
-      slot: updateRequest.storageSlot,
-      value: updateRequest.newValue,
-      counter: new Fr(updateRequest.sideEffectCounter ?? Fr.ZERO),
-    });
-
     // We need to manually populate the cache.
     ctx.persistableState.publicStorage.write(
       ctx.environment.storageAddress,
       updateRequest.storageSlot,
       updateRequest.newValue,
     );
   }
-
-  for (const nullifier of result.newNullifiers) {
-    ctx.persistableState.trace.newNullifiers.push({
-      storageAddress: ctx.environment.storageAddress,
-      nullifier: nullifier.value,
-      counter: nullifier.counter,
-    });
-  }
-
-  for (const noteHash of result.newNoteHashes) {
-    ctx.persistableState.trace.newNoteHashes.push({
-      storageAddress: ctx.environment.storageAddress,
-      noteHash: noteHash.value,
-      counter: noteHash.counter,
-    });
-  }
-
-  for (const message of result.newL2ToL1Messages) {
-    ctx.persistableState.newL1Messages.push(message);
-  }
-
-  for (const log of result.unencryptedLogs.logs) {
-    ctx.persistableState.newLogs.push(new UnencryptedL2Log(log.contractAddress, log.selector, log.data));
-  }
 }
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines 180 to 232]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Don't need to keep the trace up to date since ultimately.... It isn't really used at all right now! So for the sake of "interoperability" I say let's just stop caring about the trace being updated properly.

It is still important however that caches get updated when the AVM processes side-effects coming from the ACVM.<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/5820?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
Ok if you say so! But does it hurt to have them? If we have any test that has a nested call and we check traces then it wouldn't work? If that's not gonna happen no worries

Reply:
It's just getting hard to wrap my head around everything that we are trying to keep consistent at the AVM-ACVM boundary.

The trace is there to inform the AVM circuit and ultimately a future kernel version of what happened in the AVM. For now I think we accomplish the same thing by properly tracking the PublicExecutionResult and nestedExecutions. So maybe it's okay not to properly accomplish "AVM-ACVM interoperability" at the trace level.

Reply:
That being said, if we are using the trace's storage writes to update/initialize the ACVM's stateDb, then maybe it _does_ need to be up to date :sweat: